rust/template/differential_datalog/test.rs

```diff
@@ -639,20 +376,16 @@ fn test_semijoin(nthreads: usize) {
             })))),
         }
     };
-    fn fmfun1(v: Value) -> Option<Value> {
-        match v {
-            Value::Tuple2(v1, _v2) => Some(*v1),
-            _ => None,
-        }
+    fn fmfun1(v: DDValue) -> Option<DDValue> {
+        let Tuple2(ref v1, ref _v2) = unsafe { Tuple2::<U64>::from_ddvalue(v) };
```

Comment:
This does not seem to be the same thing as the original code - this can never return None.
Was the original code unable to statically typecheck? 
Where is the dynamic type check made?

Reply:
The original code used an `enum Value` and had to always match on it even though we knew exactly what variant we handle in every function. Now we use unsafe conversion instead.